list-rpm-versions

```diff
@@ -0,0 +1,268 @@
+directory, as found in output-NNN/output/osg-test-*.log
+
+The output argument can also be a root.log from a koji/mock build,
+or the raw output of an 'rpm -qa' command.
+
+If any packages are specified, limit the results to just those packages.
+
+If a run directory (or, just the timstamp string) is specified, summary
+information will be printed for the listed packages across all output-NNN
+subdirectories for that set of osg test runs.
+
+Options:
+  -A, --no-strip-arch  don't attempt to strip .arch from package names
+  -D, --no-strip-dist  don't attempt to strip .dist tag from package releases
+
+  -s, --summarize      summarize results for all output subdirs
+  -l, --list-outputs   list output numbers (summarize mode only)
+  -L, --max-outputs N  list at most N output numbers per NVR (-1 for unlimited)
+"""
+
+import collections
+import getopt
+import glob
+import stat
+import sys
+import os
+import re
+
+
+GLOBAL_RUNS_DIR = "/osgtest/runs"
+
+outdir     = None
+pkgs       = []
+strip_arch = True
+strip_dist = True
+summarize  = False
+list_nums  = False
+max_nums   = 7
+
+arch_pat = r'\.(x86_64|i[3-6]86|noarch|src)$'
+dist_pat = r'((\.osg(\d+)?)?\.[es]l[5-9](_[\d.]+)?(\.centos)?|\.osg|\.fc\d+)$'
+
+def usage(msg=None):
+    if msg:
+        print "***", msg, "***"
+    print __doc__ % {"script": os.path.basename(__file__)}
+    sys.exit()
+
+def parseargs():
+    global outdir, pkgs, strip_arch, strip_dist, summarize, list_nums, max_nums
+    longopts = ['no-strip-arch', 'no-strip-dist', 'summarize',
+                'list-outputs', 'max-outputs=', 'help']
+    ops,args = getopt.getopt(sys.argv[1:], 'ADslL:', longopts)
+    for op,val in ops:
+        if   op in ('-A', '--no-strip-arch') : strip_arch = False
+        elif op in ('-D', '--no-strip-dist') : strip_dist = False
+        elif op in ('-s', '--summarize')     : summarize  = True
+        elif op in ('-l', '--list-outputs')  : list_nums  = True
+        elif op in ('-L', '--max-outputs')   : list_nums  = True; \
+                                               max_nums   = int(val)
+        elif op == '--help'                  : usage()
+
+    if not args:
+        usage("Must provide a test run output location")
+
+    outdir = args[0]
+    pkgs   = args[1:]
+
+    if max_nums < 0:
+        max_nums = 99999
+
+    if summarize and not pkgs:
+        usage("Must specify package list for --summarize")
+
+def arch_strip(na):
+    return re.sub(arch_pat, '', na)
+
+def dist_strip(evr):
+    ev,r = evr.split('-')
+    r = re.sub(dist_pat, '', r)
+    return '-'.join([ev,r])
+
+def nvrgen(items):
+    # generate sequence of ["name.arch", "epoch:version-release"] pairs
+    while items:
+        na, evr = items[:2]
+        if strip_arch:
+            na = arch_strip(na)
+        if strip_dist:
+            evr = dist_strip(evr)
+        if evr.startswith("0:"):
+            evr = evr[2:]
+        yield [na,evr]
+        items[:2] = []
+
+def isdir(fn):
```

Comment:
Ah, looks like this started out in `compare-rpm-versions` as an [`isfile` function](https://github.com/edquist/tools/blob/7d49e71b633c30a03e195e37d6abfd1694c17feb/compare-rpm-versions#L71-L75), which probably there wasn't an equivalent of in `os.path` ... when I [changed the logic](
https://github.com/edquist/tools/commit/09275a443944997301ab16632a5027793db8b700#diff-1dbfd2157db6e355574163e895b4585eR75) to use `isdir` I made the obvious change without realizing there was an `os.path.isdir` that did the same thing.